src/test/HostActivationTests/NativeHosting/HostContext.cs

```diff
@@ -224,166 +160,368 @@ public void GetDelegate_Multiple(string checkProperties)
-                        .And.FailToSetRuntimePropertyValue(LogPrefix.Secondary, sharedState.ConfigPropertyName, InvalidArgFailure)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, PropertyValueFromHost)
-                        .And.HavePropertyMock(sharedState.ConfigPropertyName, PropertyValueFromHost);
+                        .InitializeContextForApp(appOrConfigPath)
+                        .And.ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
-                case CheckProperties.Remove:
+                case Scenario.NonContextMixed:
                     result.Should()
-                        .SetRuntimePropertyValue(LogPrefix.App, sharedState.AppPropertyName)
-                        .And.SetRuntimePropertyValue(LogPrefix.App, sharedState.ConfigPropertyName)
-                        .And.FailToSetRuntimePropertyValue(LogPrefix.Secondary, sharedState.AppPropertyName, InvalidArgFailure)
-                        .And.FailToSetRuntimePropertyValue(LogPrefix.Secondary, sharedState.ConfigPropertyName, InvalidArgFailure)
-                        .And.NotHavePropertyMock(sharedState.AppPropertyName)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
-                case CheckProperties.GetAll:
+            }
+
+            if (isCompatibleVersion.HasValue)
+            {
+                if (isCompatibleVersion.Value)
+                {
+                    result.Should().Pass()
+                        .And.InitializeSecondaryContext(frameworkCompatConfig, CoreHostAlreadyInitialized);
+                }
+                else
+                {
+                    result.Should().Fail()
+                        .And.FailToInitializeContextForConfig(CoreHostIncompatibleConfig)
+                        .And.HaveStdErrMatching($".*The specified framework '{frameworkName}', version '{version}', apply_patches=[0-1], roll_forward=[^ ]* is incompatible with the previously loaded version '{SharedTestState.NetCoreAppVersion}'.*");
+                }
+            }
+            else
+            {
+                result.Should().Fail()
+                    .And.FailToInitializeContextForConfig(CoreHostIncompatibleConfig)
+                    .And.HaveStdErrContaining($"The specified framework '{frameworkName}' is not present in the previously loaded runtime");
+            }
+        }
+
+        [Theory]
+        [InlineData(Scenario.ConfigMultiple,  SharedTestState.ConfigPropertyName, SharedTestState.ConfigPropertyValue,   SharedTestState.ConfigPropertyValue)]
+        [InlineData(Scenario.ConfigMultiple,  SharedTestState.ConfigPropertyName, "NEW_PROPERTY_VALUE",                  SharedTestState.ConfigPropertyValue)]
+        [InlineData(Scenario.ConfigMultiple,  "NEW_PROPERTY_NAME",                "NEW_PROPERTY_VALUE",                  null)]
```

Comment:
Please add a test with empty property values and also another one where the property value difference is small:
* One is a substring of the other
* They only differ in casing

Ideally also a test with multiple properties in both configs..., but that is probably trickier to setup.

Reply:
Made it so that there are cases with multiple properties. The set up I ended up with ('multiple' is really just two) is a little awkward though.